integrations/integrations_test.go

```diff
@@ -0,0 +1,187 @@
+package integrations
+
+import (
+	"context"
+	"io"
+	"net"
+	"strings"
+	"testing"
+	"time"
+
+	"cosmossdk.io/log"
+	"cosmossdk.io/store/snapshots"
+	storetypes "cosmossdk.io/store/types"
+	abcitypes "github.com/cometbft/cometbft/abci/types"
+	bftclient "github.com/cometbft/cometbft/rpc/client/http"
+	bfttypes "github.com/cometbft/cometbft/types"
+	dbm "github.com/cosmos/cosmos-db"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/server"
+	"github.com/cosmos/cosmos-sdk/server/api"
+	serverconfig "github.com/cosmos/cosmos-sdk/server/config"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+	"github.com/cosmos/gogoproto/grpc"
+	testapp "github.com/polymerdao/monomer/testapp"
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/require"
+)
+
+// Application Constructor `appCreator` for testing
+func mockAppCreator(
+	_ log.Logger,
+	db dbm.DB,
+	_ io.Writer,
+	_ servertypes.AppOptions,
+) servertypes.Application {
+	app, err := testapp.New(db, "1")
+	if err != nil {
+		panic(err)
+	}
+
+	return &WrappedTestApp{App: app}
+}
+
+// Unit test for Monomer's custom `StartCommandHandler` callback
+func TestStartCommandHandler(t *testing.T) {
+	svrCtx := server.NewDefaultContext()
+
+	// This flag must be set, because by default it's set to ""
+	svrCtx.Viper.Set("minimum-gas-prices", "0.025stake")
+	// This flag must be set to load the Monomer genesis file
+	viper.Set("monomer-genesis-path", "./testdata/genesis.json")
+
+	clientCtx := client.Context{}
+	inProcessConsensus := true
+	opts := server.StartCmdOptions{
+		DBOpener: func(rootDir string, backendType dbm.BackendType) (dbm.DB, error) {
+			return dbm.NewMemDB(), nil
+		},
+	}
+
+	cmtListenAddr := svrCtx.Config.RPC.ListenAddress
+	cmtListenAddr = strings.TrimPrefix(cmtListenAddr, "tcp://")
+
+	go func() {
+		err := StartCommandHandler(svrCtx, clientCtx, mockAppCreator, inProcessConsensus, opts)
```

Comment:
This is a good question. The way I understand it is that we don't need them because our `environment.Env` handles cleanup. We also have to abide by the signature for `StartCommandHandler` in order to use it in the Cosmos SDK. Is this correct? cc @joshklop

Reply:
`StartCommandHandler` is kind of like a `main` function, so it is responsible for setup and shutdown.

I'm not saying I agree with this design, but as @phasewalk1 mentioned, we have to dance to the SDK's tune.